debug-ui/app/components/Orderbook.tsx

```diff
@@ -17,33 +16,56 @@ const Orderbook = ({
 }: {
   marketAddress: string;
 }): ReactElement => {
+  const [marketData, setMarketData] = useState<Buffer>();
   const [bids, setBids] = useState<RestingOrder[]>([]);
   const [asks, setAsks] = useState<RestingOrder[]>([]);
   const [currentSlot, setCurrentSlot] = useState<number>(0);
 
   const { connection: conn } = useConnection();
   const { wallet } = useWallet();
 
+  conn.onAccountChange(
```

Comment:
much nicer approach. especially since we don't care about reliability so much here.

Reply:
you need to wrap this inside an effect, you cant just open a new subscription on every component render

Reply:
also this subscription will persist on the global connection object that you pull in via useConnection if you switch markets